src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -201,6 +201,11 @@ internal virtual bool IsLabelsScopeBinder
             }
         }
 
+        /// <summary>
+        /// True if this is the top-level binder for a local function or lambda.
```

Comment:
Is it worth adding clarification for how logical lambdas, like LINQ, operate here?